File: 프로그래머스/lv2/12913. 땅따먹기/땅따먹기.java
class Solution {
    int solution(int[][] land) {
        int answer = 0;
 
        for(int i = 1; i < land.length; i++) {
            land[i][0] += max(land[i-1][1], land[i-1][2], land[i-1][3]);
            land[i][1] += max(land[i-1][0], land[i-1][2], land[i-1][3]);
            land[i][2] += max(land[i-1][1], land[i-1][3], land[i-1][0]);
            land[i][3] += max(land[i-1][1], land[i-1][2], land[i-1][0]);
        }

        for(int number : land[land.length - 1]) {
            answer = Math.max(answer, number);
        }

        return answer;
    }
 
    public int max(int a, int b, int c) {
        return Math.max(Math.max(a, b), c);
    }
}

File: 프로그래머스/2/12913. 땅따먹기/땅따먹기.py
def solution(land):
    for i in range(1, len(land)):
        land[i][0] += math_max(land[i-1][1], land[i-1][2], land[i-1][3])
        land[i][1] += math_max(land[i-1][0], land[i-1][2], land[i-1][3])
        land[i][2] += math_max(land[i-1][0], land[i-1][1], land[i-1][3])
        land[i][3] += math_max(land[i-1][0], land[i-1][1], land[i-1][2])

    return max(land[-1])

def math_max(a,b,c):
    max_value = max(max(a,b),c)
    return max_value